tensorzero-core/src/observability/mod.rs

```diff
@@ -1197,8 +1198,26 @@ pub async fn setup_observability_with_exporter_override<T: SpanExporter + 'stati
 }
 
 /// Set up Prometheus metrics exporter
-pub fn setup_metrics() -> Result<PrometheusHandle, Error> {
-    let metrics_handle = PrometheusBuilder::new().install_recorder().map_err(|e| {
+pub fn setup_metrics(metrics_config: Option<&MetricsConfig>) -> Result<PrometheusHandle, Error> {
+    let mut builder = PrometheusBuilder::new();
+
+    if let Some(config) = metrics_config
+        && let Some(buckets) = &config.inference_overhead_histogram_buckets
+    {
+        use metrics_exporter_prometheus::Matcher;
+        builder = builder
+            .set_buckets_for_metric(
+                Matcher::Full("tensorzero_inference_latency_overhead_seconds".to_string()),
+                buckets,
+            )
+            .map_err(|e| {
+                Error::new(ErrorDetails::Observability {
+                    message: format!("Failed to set histogram buckets: {e}"),
+                })
+            })?;
+    }
```

Comment:
Let's make this a separate metric rather than overloading (i.e. report both if buckets are configured):

```
tensorzero_inference_latency_overhead_seconds_histogram
```

Overloading is tricky because a fleet of gateways might brick observability, etc. This way we always have one and optionally have the other.